src/AppCommon/ServiceControl/ServiceControlAuditsByBinarySearch.cs

```diff
@@ -154,10 +154,12 @@ async Task<AuditBatch> GetAuditBatch(string endpointName, int page, int pageSize
         {
             var pathAndQuery = $"/endpoints/{endpointName}/messages/?page={page}&per_page={pageSize}&sort=processed_at&direction=desc";
 
-            var arr = await primary.GetData<JArray>(pathAndQuery, cancellationToken);
-
-            var processedAtValues = arr.Select(token => token["processed_at"].Value<DateTime>()).ToArray();
+            var arr = await primary.GetData<JsonArray>(pathAndQuery, cancellationToken);
 
+            //var processedAtValues = arr.Select(token => token["processed_at"].Value<DateTime>()).ToArray();
+#pragma warning disable CS8632 // The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
```

Comment:
Did you try adding `#nullable enable` at the top of the file or was it a can of worms?

Reply:
I tried it in another file and it was a can of worms which probably put me off it in the rest of the files. :-)
I've just tried it here though and only two other changes had to be made so I've pushed that.